File: Others/easy/BinarySearch.py
def binarySearch(array, target):
    left = 0
    right = len(array) - 1

    while left <= right:

        mid = left + int((right - left) / 2)

        if array[mid] == target:
            return mid

        if array[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1

File: Golang/easy/binary-search.go
package main

func BinarySearch(array []int, target int) int {
	left, right := 0, len(array)-1
	var mid int

	for left <= right {
		mid = left + (right-left)/2
		if array[mid] == target {
			return mid
		} else if array[mid] > target {
			right = mid - 1
		} else {
			left = mid + 1
		}
	}

	return -1
}
